mqtt-ds/src/main/java/org/apache/rocketmq/mqtt/ds/retain/RetainedMsgClient.java

```diff
@@ -0,0 +1,210 @@
+
+        option.put("firstTopic", firstTopic);
+        option.put("topic", topic);
+
+        logger.debug("GetRetainedMsgsFromTrie option:" + option);
+
+        final ReadRequest request = ReadRequest.newBuilder().setGroup(GROUPNAME + GROUP_SEQ_NUM_SPLIT + "0").setOperation("trie").setType(Constants.READ_INDEX_TYPE).putAllExtData(option).build();
+
+        CLICLIENTSERVICE.getRpcClient().invokeAsync(leader.getEndpoint(), request, new InvokeCallback() {
+            @Override
+            public void complete(Object result, Throwable err) {
+                if (err == null) {
+                    Response rsp = (Response) result;
+                    if (!rsp.getSuccess()) {
+                        logger.info("GetRetainedTopicTrie failed. {}", rsp.getErrMsg());
+                        return;
+                    }
+                    List<ByteString> datalistList = rsp.getDatalistList();
+                    ArrayList<Message> resultList = new ArrayList<>();
+                    for (ByteString tmp : datalistList) {
+                        resultList.add(JSON.parseObject(tmp.toStringUtf8(), Message.class));
+                    }
+                    future.complete(resultList);
+                    logger.debug("-------------------------------GetRetainedTopicTrie success.----------------------------------");
+                } else {
+                    logger.debug("-------------------------------GetRetainedTopicTrie fail.-------------------------------------");
+                    logger.error("", err);
+                    future.complete(null);
+                }
+            }
+
+            @Override
+            public Executor executor() {
+                return null;
+            }
+        }, 5000);
+    }
+
+
+    public static void GetRetainedMsg(String topic, CompletableFuture<Message> future) throws RemotingException, InterruptedException {
+
+        HashMap<String, String> option = new HashMap<>();
+        option.put("topic", topic);
+
+        final ReadRequest request = ReadRequest.newBuilder().setGroup(GROUPNAME + GROUP_SEQ_NUM_SPLIT + "0").setOperation("topic").setType(Constants.READ_INDEX_TYPE).putAllExtData(option).build();
+
+        CLICLIENTSERVICE.getRpcClient().invokeAsync(leader.getEndpoint(), request, new InvokeCallback() {
+
+            @Override
+            public void complete(Object result, Throwable err) {
+                if (err == null) {
+                    Response rsp = (Response) result;
+                    if (!rsp.getSuccess()) {
+                        logger.info("GetRetainedMsg failed. {}", rsp.getErrMsg());
+                        return;
+                    }
+                    if (rsp.getData().toStringUtf8().equals("null")) {  //this topic doesn't exist retained msg
+                        return;
+                    }
+                    String strMsg = rsp.getData().toStringUtf8();
+                    Message message = JSON.parseObject(strMsg, Message.class);
```

Comment:
Is there a better way for Convert Java object to protobuf or parseObject from protobuf？